src/platform_impl/ios/app_state.rs

```diff
@@ -618,3 +619,54 @@ impl EventLoopWaker {
         }
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct Capabilities {
+    pub safe_area: bool,
+    pub home_indicator_hidden: bool,
+    pub defer_system_gestures: bool,
+    pub maximum_frames_per_second: bool,
+    pub force_touch: bool,
+}
+
+impl NSOperatingSystemVersion {
+    fn meets_requirements(&self, required_major: NSInteger, required_minor: NSInteger) -> bool {
+        (self.major, self.minor) >= (required_major, required_minor)
+    }
+}
+
+impl From<NSOperatingSystemVersion> for Capabilities {
+    fn from(os_version: NSOperatingSystemVersion) -> Capabilities {
+        assert!(
+            os_version.major >= 8,
+            "`winit` requires iOS version 8 or greater"
+        );
+
+        let safe_area = os_version.meets_requirements(11, 0);
```

Comment:
It would be good to have comments documenting the source for all of these requirements.